src/Node/ImportNode.php

```diff
@@ -22,18 +22,28 @@
  */
 class ImportNode extends Node
 {
-    public function __construct(AbstractExpression $expr, AbstractExpression $var, int $lineno, string $tag = null)
+    public function __construct(AbstractExpression $expr, AbstractExpression $var, int $lineno, string $tag = null, bool $global = true)
     {
-        parent::__construct(['expr' => $expr, 'var' => $var], [], $lineno, $tag);
+        parent::__construct(['expr' => $expr, 'var' => $var], ['global' => $global], $lineno, $tag);
     }
 
     public function compile(Compiler $compiler)
     {
+        $compiler->addDebugInfo($this);
+        if ($this->getAttribute('global')) {
+            $compiler
+                ->write('$macros[')
+                ->repr($this->getNode('var')->getAttribute('name'))
+                ->raw('] = ')
+                ->raw('$this->')
+            ;
+        } else {
+            $compiler->write('$');
+        }
         $compiler
-            ->addDebugInfo($this)
-            ->write('')
-            ->subcompile($this->getNode('var'))
-            ->raw(' = ')
+            ->raw('macros[')
```

Comment:
why not making the code look like that instead:

```php
$compiler
    ->addDebugInfo($this)
    ->write('$macros[')
    ->repr($this->getNode('var')->getAttribute('name'))
    ->raw('] = ')
;

if ($this->getAttribute('global')) {
    $compiler
        ->raw('$this->macros[')
        ->repr($this->getNode('var')->getAttribute('name'))
        ->raw('] = ')
    ;
}
```

That would be more readable that this magic with `$` or `$this->` before that code, as we would be adding the global part after the local one instead of repurposing the local one into the global one after having duplicated the local one.

The generated code would be the same.

Reply:
done